app/src/features/Columns/ColumnSelect.tsx

```diff
@@ -61,14 +65,28 @@ const ColumnSelects = ({
       skip: !credentials?.[0],
     }
   );
+  const { data: mapping } = useApiResourcesRetrieveQuery(
+    { id: mappingId ?? "" },
+    { skip: !mappingId }
+  );
 
-  const { table, column, owner: ownerId } = pendingColumn;
-  const selectedOwner = credentialOwners?.find(({ id }) => id === ownerId);
+  const { table: pendingColumnTable, column, owner: ownerId } = pendingColumn;
+  const table = (pendingColumnTable || mapping?.primary_key_table) ?? undefined;
+  const selectedOwner = ownerId
+    ? credentialOwners?.find(({ id }) => id === ownerId)
+    : credentialOwners?.find(({ id }) => id === mapping?.primary_key_owner);
   const schema = selectedOwner?.schema as Record<string, string[]>;
-  const defaultValue = {
-    id: "/",
-    label: "/",
-  };
+  const defaultValue: { id: string; label: string } =
```

Comment:
I don't really get this, why do you need to change `defaultValue`?

Reply:
If there is a mapping, it already has a primary key table and and primary key owner, so you don't need to add a default value with nothing inside ( `{ id:"/", label:"/" }` ), because you need to prefill the "table" select
But if you don't have a primary key table or a primary key owner (for exemple when you create a mapping), you still need to have a default value with nothing inside.

Reply:
i don't need it